### ISSUE TITLE
Move all of our wiki pages into the /docs folder and change inbound links

### ISSUE THREAD

**Issue by troessner** [troessner/reek#482]
Like discussed in #479

**Comment by mvz** [comment]
Perhaps we can generate those docs from the code comments of each smell?

**Comment by troessner** [comment]
>> Perhaps we can generate those docs from the code comments of each smell?

Why would we do that?
Our wiki pages are pretty good and we can migrate them automatically since the wiki is a git repo as well.

**Comment by troessner** [comment]
Started to work on that in #494

**Comment by mvz** [comment]
What I was getting at is having only one explanation of each smell, and generating all other renderings from that. However, our wiki pages are probably much more elaborate than the current smell detector comments. So perhaps we can refer from those comments to the information in `doc/`.

**Comment by troessner** [comment]
>> So perhaps we can refer from those comments to the information in doc/

You mean just like "Please see doc/Attribute.md for details" or something more elaborate?

**Comment by mvz** [comment]
Yeah, basically, you would get something like:
```
# Detects the Attribute smell. Please see doc/Attribute.md for details.
class Attribute
  # ...
end
```

**Comment by troessner** [comment]
Good idea! But let's do that in a follow-up pull request since it has nothing to do with the original pull request (we could have implemented your suggestion with the wiki structure as well). 
Additionally we should then do this for all files in docs/. That means add this comment to our spec integration, to our rake task, and so on.....

I created a follow-up issue #499 for this.

**Comment by troessner** [comment]
Fixed in #509
